src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts

```diff
@@ -1,159 +1,237 @@
+      return {
+        token0Decimals,
+        token1Decimals,
+        token0Reserves,
+        token1Reserves,
+        token0Address,
+        token1Address,
+        apr,
+      }
+    })()
+
+    const { chainId } = fromAssetId(opportunityId)
+    const token0MarketData: MarketData = selectMarketDataById(
+      state,
+      token0Address === WETH_TOKEN_CONTRACT_ADDRESS
+        ? ethAssetId
+        : toAssetId({
+            assetNamespace: 'erc20',
+            assetReference: token0Address,
+            chainId,
+          }),
     )
-  })
 
-  const data = {
-    byId: {
-      [opportunityId]: {
-        apy,
-        assetId: opportunityId,
-        id: opportunityId,
-        provider: DefiProvider.UniV2,
-        tvl,
-        type: DefiType.LiquidityPool,
-        underlyingAssetId: opportunityId,
-        underlyingAssetIds,
-        underlyingAssetRatiosBaseUnit: [
-          toBaseUnit(token0PoolRatio.toString(), pair.token0.decimals ?? 18),
-          toBaseUnit(token1PoolRatio.toString(), pair.token1.decimals ?? 18),
-        ] as const,
-        name: `${assets.byId[underlyingAssetIds[0]]?.symbol}/${
-          assets.byId[underlyingAssetIds[1]]?.symbol
-        } Pool`,
-        rewardAssetIds: [],
-        isClaimableRewards: false,
-      },
-    },
+    const assetId0 =
+      // Uniswap uses ERC20 WETH under the hood, but we want to display it as ETH
+      token0Address === WETH_TOKEN_CONTRACT_ADDRESS
+        ? ethAssetId
+        : toAssetId({
+            assetNamespace: 'erc20',
+            assetReference: token0Address,
+            chainId,
+          })
+    // Uniswap uses ERC20 WETH under the hood, but we want to display it as ETH
+    const assetId1 =
+      token1Address === WETH_TOKEN_CONTRACT_ADDRESS
+        ? ethAssetId
+        : toAssetId({
+            assetNamespace: 'erc20',
+            assetReference: token1Address,
+            chainId,
+          })
+    const underlyingAssetIds = [assetId0, assetId1] as const
+    const lpAsset = assets.byId[opportunityId]
+
+    if (bnOrZero(token0MarketData?.price).isZero()) {
+      continue
+    }
+
+    const token0Price = token0MarketData.price
+
+    const { assetReference } = fromAssetId(opportunityId)
+    // Checksum
+    const contractAddress = ethers.utils.getAddress(assetReference)
+    const uniV2LPContract = getOrCreateContractByType({
+      address: contractAddress,
+      type: ContractType.UniV2Pair,
+    })
+    const apy = bnOrZero(apr).div(100).toString()
 
+    // Getting the ratio of the LP token for each asset
+    const totalSupplyBaseUnit =
+      zapperAppBalanceData?.supply && lpAsset
+        ? bnOrZero(zapperAppBalanceData.supply).times(bn(10).pow(lpAsset.precision)).toString()
+        : (await uniV2LPContract.totalSupply()).toString()
+
+    const token0ReservesBaseUnit = bnOrZero(
+      bnOrZero(bnOrZero(token0Reserves.toString()).toString()),
+    ).times(bn(10).pow(lpAsset?.precision ?? 18))
```

Comment:
let's not opt chain precisions if we don't have lp asset bail

Reply:
Noice catch - 4536846e9d7771a64012c5d3fff8313b7089b016